pysph/tools/ipython.py

```diff
@@ -0,0 +1,641 @@
+            display(*imgs)
+        else:
+            print("No results to show.")
+
+    def show_info(self):
+        '''
+        Print contents of the .info file present in the output directory,
+        keys present in results.npz,  number of files and
+        information about paricle arrays.
+
+        Examples
+        --------
+
+        >>> sample = Viewer2D('/home/deep/pysph/trivial_inlet_outlet_output/')
+        >>> sample.show_info()
+        Printing info :
+
+        cpu_time : 2.58390402794
+        args : []
+        completed : True
+        fname : trivial_inlet_outlet
+        output_dir : /home/deep/pysph_coverage/pysph_coverage/trivial_inlet_outlet_output
+        Number of files : 31
+          inlet :
+            Number of particles : 55
+            Output Property Arrays : ['x', 'y', 'z', 'u', 'v', 'w', 'rho',
+            'm', 'h', 'pid', 'gid', 'tag']
+          fluid :
+            Number of particles : 0
+            Output Property Arrays : ['x', 'y', 'z', 'u', 'v', 'w', 'rho',
+             'm', 'h', 'pid', 'gid', 'tag']
+          outlet :
+            Number of particles : 0
+            Output Property Arrays : ['x', 'y', 'z', 'u', 'v', 'w', 'rho',
+            'm', 'h', 'pid', 'gid', 'tag']
+
+        Keys in results.npz :
+        ['t', 'x_max']
+        '''
+        # Imports #
+
+        import json
+        import glob
+        from pysph.solver.utils import load
+
+        # General Info #
+
+        path = self.path + "*.info"
+        infofile = open(glob.glob(path)[0], 'r')
+        data = json.load(infofile)
+        infofile.close
+        print('Printing info : \n')
+        for key in data.keys():
+            if key == 'cpu_time':
+                print(key + " : " + str(data[key]) + " seconds")
+            else:
+                print(key + " : " + str(data[key]))
+
+        print('Number of files : {}'.format(len(self.paths_list)))
+
+        # Particle Info #
+
+        temp_data = load(self.paths_list[0])['arrays']
+
+        for key in temp_data:
+            print("  {} :".format(key))
+            print("    Number of particles : {}".format(
+                temp_data[key].get_number_of_particles())
+            )
+            print("    Output Property Arrays : {}".format(
+                temp_data[key].output_property_arrays)
+            )
+
+        # keys in results.npz
+
+        import glob
+        from numpy import load as npl
+
+        path = self.path + "*results*"
+        files = glob.glob(path)
+        if len(files) != 0:
+            data = npl(files[0])
+            print("\nKeys in results.npz :")
+            print(data.keys())
+
+    def show_all(self):
+        self.show_info()
+        self.show_results()
+        self.show_log()
+
+
+class Viewer2D(Viewer):
+
+    def _create_widgets(self):
+
+        from pysph.solver.utils import load
+        import ipywidgets as widgets
+
+        temp_data = load(self.paths_list[0])['arrays']
+        self._widgets = {}
```

Comment:
I did that so that I can create a hierarchy such as widgets -> particles -> boundary -> scalar_property

I can create another object called `ViewerWidgets` but then again, it'll be like so : 
`self.widgets = ViewerWidgets(path)` 
And then again I'll have to use `self.widgets['particles'][particles_type]['scalar_Scale']` and so on.